src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlExceptionTest.cs

```diff
@@ -33,28 +33,6 @@ public void SerializationTest()
             Assert.Equal(e.StackTrace, sqlEx.StackTrace);
         }
 
-
-        [Fact]
-        [ActiveIssue("12161", TestPlatforms.AnyUnix)]
-        public static void SqlExcpetionSerializationTest()
-        {
-            var formatter = new BinaryFormatter();
-            SqlException e = CreateException();
-            using (var stream = new MemoryStream())
-            {
-                try
-                {
-                    formatter.Serialize(stream, e);
-                    stream.Position = 0;
-                    var e2 = (SqlException)formatter.Deserialize(stream);
-                }
-                catch (Exception ex)
-                {
-                    Assert.False(true, $"Unexpected Exception occurred: {ex.Message}");
-                }
-            }
-        }
-
```

Comment:
This is something that we need to spend more time on. Supported serialization on NetFx is [DataContractSerializer](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datacontractserializer?view=net-5.0) or [XMLSerializer](https://docs.microsoft.com/en-us/dotnet/api/system.xml.serialization.xmlserializer?view=net-5.0). Json is not in netfx. XMLSerializer only works on public members and cannot work on ICollections apparently. Our only option is DataContractSerializer and that requires breaking changes on the SqlException class, such as attributes. A class cannot have [Serializable] and [DataContract] at the same time. that is the main issue I hit while I was working on it. However, .NET 5 compalins about it in our test. it could be marked off for netcore and still get tested or since binaryFormatter is not supported removed from our test lab as well. Both will do the job for Net 5 test labs.

Reply:
> A class cannot have [Serializable] and [DataContract] at the same time. 

Does it mean existing applications doing as above will break?
Do we have no alternative? That doesn't sound right to me.

Reply:
>> Json is not in netfx.
Maybe I misunderstand but: 
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.json.datacontractjsonserializer?view=net-5.0

Reply:
@ErikEJ Thanks for mentioning this. My source for the changes was this [MS document](https://docs.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide#preferred-alternatives), However what you mentioned here looks interesting to me. I will spend some time to test how I can fit it to our problem here. For now, we have decided to keep the test, but disable it for .NET 5 and higher versions as the error was thrown by .NET 5 runtime.

Reply:
We have decided to disable the test for Net 5 and later versions. A constant has been added to address that matter.